_source/_posts/2020-09-22-reintroducing-joel-franusic.md

```diff
@@ -0,0 +1,331 @@
+
+def a000045():
+    ring = [3, 5]
+    while True:
+        x = ring[0] + ring[1]
+        yield(x)
+        ring.append(x)
+        ring.pop(0)
+
+def a001477():
+    x = 1
+    while True:
+        yield(x)
+        x = x + 1
+
+def a016777():
+    x = 1
+    while True:
+        yield(x)
+        x = x + 3
+
+def a000079():
+    x = 1
+    while True:
+        yield(2 ** x)
+        x = x + 1
+
+def play(func):
+    document["messages"].clear()
+    document["error"].clear()
+    document["error"].style.display = "none"
+    for incrementor in [a000004, a001477, a016777, a000079, a000045]:
+        counter = incrementor()
+        ring = [counter.__next__(), counter.__next__(), counter.__next__()]
+        tests = 0
+        name = incrementor.__name__
+        checklist_message = f"""Sequence <a href="https://oeis.org/{name}" target="_blank">{name}</a>."""
+        for want in counter:
+            got = func(ring[0],ring[1],ring[2])
+            if want != got:
+                msg =  f"""Error with sequence {name}:
+                          Expected <code>answer({ring[0]}, {ring[1]}, {ring[2]})</code> to return <code>{want}</code> 
+                          but got <code>{got}</code> instead."""
+                checklist_message = f"""Sequence <a href="https://oeis.org/{name}" target="_blank">{name}</a>."""
+                document["messages"] <= html.LI(checklist_message, Class="fail")
+                document["error"] <= html.DIV(msg, Class="error")
+                document["error"].style.display = "block"
+                return False
+            ring.append(want)
+            ring.pop(0)
+            tests = tests + 1
+            if tests > 50:
+                break
+        document["messages"] <= html.LI(checklist_message, Class="pass")
+    document["success"].style.display = "block"
+    return True
+
+def runPy():
+    print("runPy called")
+    code = pyEditor.getValue()
+    exec(code)
+
+refresh_from_editor_delay_ms = 400
+
+def edit_hook(cm, *arg):
+    if cm.typing_delay_timer:
+        timer.clear_timeout(cm.typing_delay_timer)
+    cm.typing_delay_timer = timer.set_timeout(runPy, refresh_from_editor_delay_ms)
+
+def init_hook(cm):
+    cm.typing_delay_timer = None
+    cm.on("changes", edit_hook)
+    cm.on("update", edit_hook)
+
+window.CodeMirror.defineInitHook(init_hook)
+
+pyEditor = window.CodeMirror.fromTextArea(document["editor"], {
+    "lineNumbers": True,
+    "mode": "python",
+    "matchBrackets": True,
+    "indentUnit": 4,
+    "theme": "neat"
+})
+
+</script>
+
+{% img blog/reintroducing-joel-franusic/joel-franusic-ibm-729.jpg alt:"Joël Franusic standing in front of two IBM 729 reel-to-reel tape machines for the IBM 1401" width:"800" %}{: .center-image }
+
+Hi, my name is Joël Franusic and I'm happy to announce that I am,
+once again, a developer advocate at Okta, focusing on the Python and Go communities.
+
+(Why do I say "once again"? Because 6 years ago, I started my Okta career as a
```

Comment:
```suggestion
(Why do I say "once again"? Because six years ago, I started my Okta career as a
```

I learned while writing at InfoQ that all numbers below 10 should be spelled out.